packages/containers/src/Form/Form.container.js

```diff
@@ -0,0 +1,131 @@
+import React, { PropTypes } from 'react';
+import { Map } from 'immutable';
+import { componentState } from 'react-cmf';
+import ComponentForm from 'react-talend-forms';
+import classnames from 'classnames';
+
+export const DEFAULT_STATE = new Map({});
+
+/**
+ * Because we don't want to loose form input
+ * This Component bind onChange to store the formData in it's state.
+ * <Form jsonSchema={} uiSchema={} data={} />
+ */
+class Form extends React.Component {
+	static displayName = 'Container(Form)';
+	static propTypes = {
+		formId: PropTypes.string.isRequired,
+		...componentState.propTypes,
+	};
+
+	/**
+	 * return the form data in redux state
+	 * usefull in mapStateToProps of your component
+	 * @example
+	const FORM_ID = 'add-datastore-form';
+
+	 *
+	 * @param  {[type]} state  [description]
+	 * @param  {[type]} formId [description]
+	 * @return {[type]}        [description]
+	 */
+	static getFormData(state, formId) {
+		return state.cmf.components.getIn(['Container(Form)', formId, 'data']);
+	}
+
+	constructor(props) {
+		super(props);
+		this.formActions = this.formActions.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.onTrigger = this.onTrigger.bind(this);
+		this.onSubmit = this.onSubmit.bind(this);
+		this.jsonSchema = this.jsonSchema.bind(this);
+		this.uiSchema = this.uiSchema.bind(this);
+	}
+
+	onTrigger(formData, formId, propertyName, propertyValue) {
+		if (this.props.onTrigger) {
+			this.props.onTrigger(formData, formId, propertyName, propertyValue);
+		}
+		// TODO: add support for props.onTriggerActionCreator
+	}
+
+	onChange(form) {
+		this.props.setState({ data: form.formData, dirty: true });
+		if (this.props.onChange) {
+			this.props.onChange(...form);
+		}
+	}
+
+	onSubmit(formData) {
+		if (this.props.onSubmit) {
+			this.props.onSubmit(formData);
+		}
+		if (this.props.onSubmitActionCreator) {
+			this.props.dispatchActionCreator(
+				this.props.onSubmitActionCreator,
+				null,
+				{ props: this.props, formData },
+			);
+		}
+		this.props.setState({ data: undefined, dirty: false });
+	}
+
+	formActions() {
+		if (typeof this.props.actions === 'function') {
+			const state = (this.props.state || DEFAULT_STATE).toJS();
+			return this.props.actions(state.data);
+		}
+		return this.props.actions;
+	}
+
+	jsonSchema() {
+		const state = (this.props.state || DEFAULT_STATE).toJS();
+		if (typeof this.props.jsonSchema === 'function') {
+			return this.props.jsonSchema(state.data);
+		}
+		return this.props.jsonSchema;
+	}
+
+	uiSchema() {
+		const state = (this.props.state || DEFAULT_STATE).toJS();
+		if (typeof this.props.uiSchema === 'function') {
+			return this.props.uiSchema(state.data);
+		}
+		return this.props.uiSchema;
+	}
+
+	render() {
+		const state = (this.props.state || DEFAULT_STATE).toJS();
```

Comment:
There are quite a lot of repetitions of `const state = (this.props.state || DEFAULT_STATE).toJS();`.
Do you think it would be possible to minimize it ?